src/QirRuntime/lib/QIR/intrinsicsMath.cpp

```diff
@@ -28,4 +33,19 @@ double quantum__qis__arctan2__body(double y, double x)
     return std::atan2(y, x);        // https://en.cppreference.com/w/cpp/numeric/math/atan2
 }
 
+int64_t quantum__qis__drawrandomint__body(int64_t minimum, int64_t maximum)
+{
+    if(minimum > maximum)
+    {
+        throw std::invalid_argument( excStrDrawRandomInt );
```

Comment:
We should probably go through quantum__rt__fail here as the bad arguments would indicate either user or QIR generation error.

Formatting nit: we are not using spaces inside parenthesis of function calls.

Reply:
Fixed.